### PR TITLE
Fix yarn workspaces

### PR DESCRIPTION
**Pull request by filip131311**
This PR fixes node_modules detection for yarn workspaces and adds a working example of a set up in which workspaces are used. 

relates to: #514

### CODE REVIEW

File: packages/vscode-extension/src/utilities/packageManager.ts [hunk 1]
```diff
@@ -14,49 +18,74 @@ async function pathExists(p: string) {
...
+  function findWorkspace(appRoot: string) {
+    let currentDir = appRoot;
+    let parentDir = path.resolve(currentDir, "..");
+    while (parentDir !== currentDir) {
+      currentDir = parentDir;
+      parentDir = path.resolve(currentDir, "..");
+      const packageJsonPath = path.join(currentDir, "package.json");
+      if (!existsSync(packageJsonPath)) {
+        continue;
+      }
```
Comment (jakub-gonet) [inline packages/vscode-extension/src/utilities/packageManager.ts:31]: I think we should just read the file and catch if doesn't exist, something like that:
```ts
function walkDirectoriesUntilRoot(
  start: string,
  shouldStop: (current: string) => boolean
): string | undefined {
  let currentDir = start;
  let parentDir = path.resolve(currentDir, "..");
  while (parentDir !== currentDir) {
    currentDir = parentDir;
    parentDir = path.resolve(currentDir, "..");

    if (shouldStop(currentDir)) {
      return currentDir;
    }
  }
  return undefined;
}

function findWorkspace(appRoot: string) {
  return walkDirectoriesUntilRoot(appRoot, (currentDir) => {
    try {
      const packageJsonPath = path.join(currentDir, "package.json");
      const hasWorkspaces = require(packageJsonPath).workspaces !== undefined;
      return hasWorkspaces;
    } catch (error) {
      // No package.json
      return false;
    }
  });
}
```

**Review by jakub-gonet — APPROVED** [review]

File: packages/vscode-extension/src/utilities/packageManager.ts (hunk 1 above)
Comment (filip131311) [reply packages/vscode-extension/src/utilities/packageManager.ts]: i did that just reaarengedsome functions to reuse code

File: packages/vscode-extension/src/extension.ts [hunk 2]
```diff
@@ -273,31 +321,24 @@ async function findAppRootFolder() {
...
-    "**/node_modules/react-native/package.json",
-    null
-  );
-  if (rnPackageLocation) {
-    return Uri.joinPath(rnPackageLocation, "../../..").fsPath;
+  if (appRootCandidates.length > 1) {
+    const openLaunchConfigButton = "Open Launch Configuration";
+    window
+      .showErrorMessage(
+        `multiple reat-native applications were detected in the workspace, ${appRootCandidates[0]} were chosen, but you can change the root of your application in lunch configuration.`,
```
Comment (kmagiera) [inline packages/vscode-extension/src/extension.ts:330]: add some text that says that in order to get rid of this error, people should specify app root in the launch configuration

**Review by kmagiera — APPROVED** [review]

File: packages/vscode-extension/src/extension.ts (hunk 2 above)
Comment (jakub-gonet) [reply packages/vscode-extension/src/extension.ts:330]: Also, "lunch configuration" typo. 🥪

**Comment by DorianMazur** [comment]
This PR probably broke react-native-ide. Now node_modules don't work. I had to downgrade to 0.17

**Comment by filip131311** [comment]
@DorianMazur hello, thank you so much for your feedback would you mind sharing React Native IDE logs from 0.18 version with us? you can find them in the output section of your vscode. Why do you think it is this PR that broke it for you?

**Comment by DorianMazur** [comment]
I'm using yarn with corepack in a project and it basically can't find node_modules. After downgrading to 0.17 everything works like a charm.

**Pull request merged**